S_Fighter.java

```diff
@@ -614,11 +614,26 @@ public void actionPerformed(ActionEvent event) {
 			try {
 				String[] array = tf_pickup.getText().trim().split(",");
 				int array_sz = array.length;
-				item_ids = new int[array_sz];
+				ArrayList<Integer> ids = new ArrayList<Integer>();
 				banked_count = new int[array_sz];
 				has_banked = new boolean[array_sz];
-				for (int i = 0; i < array_sz; i++) {
-					item_ids[i] = Integer.parseInt(array[i]);
+				for (String a : array) {
+					if (a.contains("-")) {
+						String[] bounds = a.split("-");
+						int lower_bound = Integer.parseInt(bounds[0]);
+						int upper_bound = Integer.parseInt(bounds[1]);
+						if (lower_bound < upper_bound) {
+							for (int i = lower_bound; i <= upper_bound; i++) {
+								ids.add(i);
+							}
+						}
+					} else {
+						ids.add(Integer.parseInt(a));
+					}
+				}
+				int[] item_ids = new int[ids.size()];
+				for (int i = 0; i < ids.size(); i++) {
+					item_ids[i] = ids.get(i);
```

Comment:
it'd be neater if this whole section could be simplified, perhaps by using ArrayList.toArrray(array).